File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
n, m = map(int, input().split())

board_b = [
    'BWBWBWBW',
    'WBWBWBWB',
    'BWBWBWBW',
    'WBWBWBWB',
    'BWBWBWBW',
    'WBWBWBWB',
    'BWBWBWBW',
    'WBWBWBWB'
]

board_w = [
    'WBWBWBWB',
    'BWBWBWBW',
    'WBWBWBWB',
    'BWBWBWBW',
    'WBWBWBWB',
    'BWBWBWBW',
    'WBWBWBWB',
    'BWBWBWBW'
]

b_cnt = 2501
w_cnt = 2501

board_state = []

for _ in range(n):
    board_state.append(input())

def cnt_b(x, y):
    cnt = 0
    for i in range(8):
        for j in range(8):
            if board_state[x + i][y + j] != board_b[i][j]:
                cnt += 1

    return cnt

def cnt_w(x, y):
    cnt = 0
    for i in range(8):
        for j in range(8):
            if board_state[x + i][y + j] != board_w[i][j]:
                cnt += 1

    return cnt

for x in range(n - 7):
    for y in range(m - 7):
        b_cnt = min(cnt_b(x, y), b_cnt)
        w_cnt = min(cnt_w(x, y), w_cnt)

print(min(b_cnt, w_cnt))

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.cc
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

string board[50];
string Wchessboard[8] = {
    {"WBWBWBWB"},
    {"BWBWBWBW"},
    {"WBWBWBWB"},
    {"BWBWBWBW"},
    {"WBWBWBWB"},
    {"BWBWBWBW"},
    {"WBWBWBWB"},
    {"BWBWBWBW"}
};
string Bchessboard[8] {
    {"BWBWBWBW"},
    {"WBWBWBWB"},
    {"BWBWBWBW"},
    {"WBWBWBWB"},
    {"BWBWBWBW"},
    {"WBWBWBWB"},
    {"BWBWBWBW"},
    {"WBWBWBWB"}
};

int WstartCheck(int y, int x) {
    int cnt = 0;
    for(int i = y;i < y + 8;i++) {
        for(int j = x;j < x + 8;j++) {
            if(board[i][j] != Wchessboard[i - y][j - x])    cnt++;
        }
    }
    return cnt;
}

int BstartCheck(int y, int x) {
    int cnt = 0;
    for(int i = y;i < y + 8;i++) {
        for(int j = x;j < x + 8;j++) {
            if(board[i][j] != Bchessboard[i - y][j - x])    cnt++;
        }
    }
    return cnt;
}

int main() {
    int n, m;
    int Min = 251;
    
    cin >> n >> m;
    
    for(int i = 0;i < n;i++) {
        cin >> board[i];
    }
    
    for(int i = 0;i + 7 < n;i++) {
        for(int j = 0;j + 7 < m;j++) {
            Min = min(Min, min(BstartCheck(i, j), WstartCheck(i, j)));
        }
    }
    cout << Min << endl;
    return 0;
}